libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -1918,11 +1916,9 @@ void IRGeneratorForStatements::endVisit(MemberAccess const& _memberAccess)
 						*_memberAccess.annotation().referencedDeclaration
 					).resolveVirtual(m_context.mostDerivedContract(), super);
 
-				define(_memberAccess) << to_string(resolvedFunctionDef.id()) << "\n";
 				solAssert(resolvedFunctionDef.functionType(true), "");
 				solAssert(resolvedFunctionDef.functionType(true)->kind() == FunctionType::Kind::Internal, "");
-				if (!_memberAccess.annotation().calledDirectly)
-					m_context.addToInternalDispatch(resolvedFunctionDef);
+				assignInternalFunctionIDIfNotCalledDirectly(_memberAccess, resolvedFunctionDef);
```

Comment:
I don't fully understand why it worked without `.part("functionIdentifier")` before (it's now present in `assignInternalFunctionIDIfNotCalledDirectly()`)? Is there some shortcut in `IRVariable` that lets you skip it if there's only one part inside a variable?

Reply:
Interesting, I did not notice! My guess is that we do not have a test that does not directly call. If you call directly, the member access yul variable is defined but never referenced. Let me find such a test and base it on top of develop.

Reply:
Ah yes, this shortcut exists: If you do not specify a part, it lists all parts as a comma separated list.

Reply:
(we have tests that touch this part of the code)

Reply:
Great. Good to know.